DynatraceAppMon/Launcher.cs

```diff
@@ -572,250 +581,259 @@ private int GetDotNetFrameworkForProject(Project project)
+
+            StopLastRunningIIS();
+
+            context.log(Context.LOG_INFO + "Starting new IIS Express instance with command line: " + startProgram + " " + startArguments);
+            lastRunningIIS = LaunchProcess(startProgram, startArguments, startWorking, "vs");
+
+            // now we start the browser with the correct page
+            OpenWebBrowserWithDelay(launchURL, this.WaitForBrowserTime); 
+        }
+
+        private void LaunchWebSiteProject(Project project)
+        {
+            context.log(Context.LOG_INFO + "Launching WebSite project");
+
+            // website web application config resides beside solution, DevelopmentServerCommandLine property gives proper command line
+            string startProgram = GetProperty(project, "DevelopmentServerCommandLine", "");
+
+            StopLastRunningIIS();
+            
+            context.log(Context.LOG_INFO + "Starting new IIS Express instance with command line: " + startProgram);
+            lastRunningIIS = LaunchProcess(startProgram, "", ".", "vs");
+
+            // now we start the browser with the correct page
+            OpenWebBrowserWithDelay(GetProperty(project, "BrowseURL", ""), this.WaitForBrowserTime);
+        }
+
+        private async void OpenWebBrowserWithDelay(string launchURL, int delay)
+        {
+            await System.Threading.Tasks.Task.Delay(delay);
+            context.log(Context.LOG_INFO + "Starting web browser, url: " + launchURL);
+            System.Diagnostics.Process.Start(launchURL);
+        }
 
-                LaunchProcess(startProgram, StartArguments, null, null);
+        private void LaunchProject(Project project)
+        {
+            // we have to start the output assembly
+            // get all project related properties    
+            bool result = true;
+            string assemblyName = GetProperty(project, "AssemblyName", UNEXPECTED_DEFAULT_PROPERTY_VALUE);
+            result &= expectProperty("AssemblyName", assemblyName);
+            string projectFileName = System.IO.Path.GetDirectoryName(project.FileName);
+
+            Properties configProperties = project.ConfigurationManager.ActiveConfiguration.Properties;
+
+            string outputDir = GetProperty(configProperties, "OutputPath", UNEXPECTED_DEFAULT_PROPERTY_VALUE);
+            result &= expectProperty("OutputPath", assemblyName);
+            string startArguments = GetProperty(configProperties, "StartArguments", UNEXPECTED_DEFAULT_PROPERTY_VALUE);
+            result &= expectProperty("StartArguments", assemblyName);
+            string startWorking = GetProperty(configProperties, "StartWorkingDirectory", UNEXPECTED_DEFAULT_PROPERTY_VALUE);
+            result &= expectProperty("StartWorkingDirectory", assemblyName);
+            string startProgram = GetProperty(configProperties, "StartProgram", UNEXPECTED_DEFAULT_PROPERTY_VALUE);
+            result &= expectProperty("StartProgram", assemblyName);
+
```

Comment:
are you using valid values for 2nd param of expectProperty (assemblyName in all cases)?